source/utils/ethUtil.ts

```diff
@@ -1,106 +1,66 @@
+    const walletController = getWalletController();
+    const ethClient = walletController.account.ethClient;
 
-    const { activeNetwork, activeWallet }: IVaultState = store.getState().vault;
-    const network = activeNetwork[KeyringNetwork.Ethereum] as EthNetworkId;
-    const knownERC20Asset = activeWallet.assets.find((asset: IAssetState) => asset.contractAddress === to && asset.type === AssetType.ERC20);
+    if (gas) return parseInt(gas);
 
-    let abi: Array<any> | false;
-    if (knownERC20Asset) {
-        abi = erc20ABI;
-    } else {
-        abi = await getContractDetails(to, network);
+    if (value !== '' && to && from) {
+        try {
+            const contract = ethClient.createERC20Contract(from);
+            const gasLimitBigNumber = await contract.estimateGas.transfer(to, parseUnits(value));
+            const gasLimit = gasLimitBigNumber.toNumber();
+            return Math.floor(gasLimit * 1.5);
+        } catch(err) {
+            return 21000;
+        }
     }
 
-    return abi;
-}
-
-export const estimateGasLimitForTransfer = async ({ to, from, amount: value }: {to: string, from: string,  amount: string}) => {
-    const { activeNetwork }: IVaultState = store.getState().vault;
-    const network = activeNetwork[KeyringNetwork.Ethereum] as EthNetworkId;
-    const abi = await _getAbi({ to });
-    const web3 = _getWeb3(network);
-    const contract = new web3.eth.Contract(abi as any, to);
-
-    if(value !== '0' && value !== ''){
-        const val = parseInt(value); // transfer throws error on decimals, int amount estimate should be basically the same
-        const gasLimit = await contract.methods.transfer(to, val).estimateGas({ from });
-        return Math.floor(gasLimit * 1.5);
-    }
-    
     return 0;
 }
 
-export const estimateGasLimit = async ({ to, data }: { to: string, data: string }): Promise<number> => {
-    const { activeNetwork, activeWallet }: IVaultState = store.getState().vault;
-    const network = activeNetwork[KeyringNetwork.Ethereum] as EthNetworkId;
+export const estimateGasLimit = async ({ to, data, gas }: { to: string, data: string, gas: string }): Promise<number> => {
+    const walletController = getWalletController();
+    const ethClient = walletController.account.ethClient;
+    const { activeWallet }: IVaultState = store.getState().vault;
     const ethAsset = activeWallet?.assets.find((asset: IAssetState) => asset.type === AssetType.Ethereum);
+    const from = ethAsset.address;
 
     if (!ethAsset || !to || to?.toUpperCase().startsWith('DAG')) {
         return 0; // DAG? 
     }
     
-    const from = ethAsset.address;
-    const abi = await _getAbi({to});
-
-    // Not a contract -> 21,000 standard gasLimit
-    if (!abi) {
-        return 21000;
+    // ETH asset
+    if (gas) return parseInt(gas);
+    if (!data) return 21000;
```

Comment:
```suggestion
    // ETH asset
    if (gas) {
        return parseInt(gas);
    } 

    if (!data){
        return 21000;
    } 
```
long versions preferred.